### PR TITLE
Feature/filter feedback for url created

### PR DESCRIPTION
**Pull request by Mr-EZG**
Changes were related to GitHub issue #456. The addresses the issue of not filtering or verifying the text used to represent a url for some health authority which could lead to potential server attacks via not regulating the maximum amount of characters that could be used to represent a URL. This implementation limits the character input used to represent a URL to 8,000. This number was chosen as to allow long urls, but still help prevent the opportunity of a server attack due to an extremely long String (See conversation on GitHub issue #456). Another filter was added as to verify the format of the String in that it followed a valid URL format. This was done as an attempt to help prevent hidden characters for server attacks and help increase the likelyhood that the user had inputted a url that is more likely to point towards a valid, live website. If the String sent to be considered as a URL fails any of the two aforementioned conditions, then the URL will not be saved and the user will be notified for which condition the have failed. As a more convenient bonus, the new feature trims leading and trailing whitespaces. It should also be noted that this current implementation also trims any attempted URL to the first 8,000 characters in the event that the input would exceed 8,000 characters.

NOTE: new variables were added to the en.json file. These included "invalid_entry_message", "please_try_again_message", "proceed_message", and "entry_already_used". For more scalable use, the translation of these variables should be included. It should also be noted that I had to ignore the lint issue since it comes from the regex expression. Some escape characters might not be needed, but I this version includes a working regex as is.

Does not allow the user to input an invalid URL and limits the overall length of potential URLs to be considered to 8,000 characters.

#### Linked issues:

GitHub issue #456

This fixes the character limit and adds an additional feature as to help prevent the insertion of hidden characters for injection attacks via a regex pattern. The pattern checks that only certain characters are used and that the overall format matches how urls are defined with slight flexibility with capitolization to match the flexibility of a web browser (Such as Www.google.com is not technically valid since no domain name can have capitol letters, but a browser would still point to www.google.com) This was an important feature as keyboards on phones begin with with typing the first letter as capitalized.

#### How to test

NOTE: due to personal computer restrictions, this was only tested on an Android Emulator. Once able to recreate the results, please do see if they could be recreated on an IOS emulator.

Home-screen -> Dashboard -> Choose Health Authority -> Add trusted Source -> Add authority via URL,

Then:

Attempt to submit a url that does not meet URL format protocol. (NOTE: this does not mean that the url points to a live website, but that the attempted URL is valid and could point to live website)
Attempt to submit a url with more than 8,000 characters
Attempt to submit a url with insert hidden characters into the text-box (as a potential consequence, so may not fully work or be true)

Try to test inputting valid and non-valid urls with edge cases included (such as an empty String)
Then if see if no error pops up for the User if it does not match the aforementioned properties. The pop up messages should be designed as to force the user to notice that their url is in an unacceptable format. There should also be log messages to help verify the above.

### CODE REVIEW

**Comment by summetj** [comment]
Hi @MrEZG, Can you please change the description to include the "fixes" keyword in front of #456 (e.g. "fixes #456") so that github will correctly link to the issue?

Also, please resolve the conflicts and remove the lint errors.

File: app/views/ChooseProvider.js [hunk 1]
```diff
@@ -46,9 +46,13 @@ class ChooseProviderScreen extends Component {
     this.state = {
       selectedAuthorities: [],
       displayUrlEntry: 'none',
+      textLimit: 8000,
       urlEntryInProgress: false,
       urlText: '',
       authoritiesList: [],
+      regexExpressionForURL: new RegExp(
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:53]: Put this regexp as a const up the top of the file. it never changes.


```js
const URL_REGEX = ...
```

File: app/views/ChooseProvider.js [hunk 2]
```diff
@@ -46,9 +46,13 @@ class ChooseProviderScreen extends Component {
     this.state = {
       selectedAuthorities: [],
       displayUrlEntry: 'none',
+      textLimit: 8000,
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:49]: use a module level const, it never changes.

File: app/views/ChooseProvider.js [hunk 3]
```diff
@@ -149,13 +153,76 @@ class ChooseProviderScreen extends Component {
...
+    this.setState({
+      urlText: text,
+    });
+  }
+
   addCustomUrlToState(urlInput) {
-    if (urlInput === '') {
-      console.log('URL input was empty, not saving');
+    if (!urlInput.match(this.state.regexExpressionForURL)) {
+      console.log('URL did not meet URL format, not saving');
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:219]: remove console .logs

File: app/views/ChooseProvider.js [hunk 4]
```diff
@@ -149,13 +153,76 @@ class ChooseProviderScreen extends Component {
     }
   }
 
+  createInvalidTextPopUp() {
+    Alert.alert(
+      languages.t('label.invalid_entry_message'),
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:158]: move all these new label.x strings to authorities.x

File: app/views/ChooseProvider.js [hunk 5]
```diff
@@ -149,13 +153,76 @@ class ChooseProviderScreen extends Component {
     }
   }
 
+  createInvalidTextPopUp() {
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:156]: Can these alerts be moved outside of the class into regular functions? they don't appear to use `this`.

File: app/views/ChooseProvider.js [hunk 6]
```diff
@@ -149,13 +153,76 @@ class ChooseProviderScreen extends Component {
...
+  }
+
+  updateUrlText(text) {
+    text = text.trim();
+    if (text.length >= this.state.textLimit) {
+      text = text.substring(0, this.state.textLimit);
+      this.entryExceededCharacterLimitPopUp();
+      console.log(
+        'URL must be ' +
+          this.state.textLimit +
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:208]: remove console.logs

File: app/locales/en.json [hunk 7]
```diff
@@ -46,6 +48,7 @@
     "import_step_3": "Visit Google Takeout and export your Location History using following settings: \n1. Delivery method: \"Add to Drive\" \n2. Frequency: \"Export once\" \n3. File type & size: \".zip\" and \"1GB\" \n4. Google sends an email when the export is ready \n5. Return here to import locations. Import options: \n- Import from Google Drive \n- Download from browser, then import from local phone files. Make sure to be on WiFi network as files can be big.",
     "import_takeout": "Visit Google Takeout",
     "import_title": "Import Locations",
+    "invalid_entry_message": "Invalid Entry",
```
Comment (tstirrat) [inline app/locales/en.json:51]: "invalid url" is more clear.

File: app/views/ChooseProvider.js [hunk 8]
```diff
@@ -220,6 +287,8 @@ class ChooseProviderScreen extends Component {
     );
   }
 
+  //limitTextLength(text, count) // Being added
```
Comment (tstirrat) [inline app/views/ChooseProvider.js:290]: remove dead code

**Comment by tstirrat** [comment]
Hello, closing due to no activity in 11 days. This is still a helpful change, so feel free to reopen when you've synced and addressed comments!

**Comment by Mr-EZG** [comment]
Hi, please see "Feature/updated url filter #791" under the new Path-Check repository for the new changes!
